Akla/Extensions/UIButton+Extension.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  UIButton+Extension.swift
+//  Akla
+//
+//  Created by Ahmad Ayman Mansour on 18/05/2023.
+//
+
+import UIKit
+
+//MARK: - StyledButton (UIButton Protocol)
+protocol StyledButton {
+    func applyButtonStyle (backgroundColor: UIColor, title: String, textColor: UIColor?)
+}
+
+
+// MARK: - UIButton Extension for StyledButton
+extension UIButton: StyledButton {
+    func applyButtonStyle(backgroundColor: UIColor, title: String, textColor: UIColor?) {
```

Comment:
You need to separate styling from the text updating logic.

Reply:
⬆️ 

`func applyButtonStyle(backgroundColor: UIColor, textColor: UIColor? = .white) {`